documentation/AuthServer.md

```diff
@@ -34,7 +35,9 @@ The document types used by the authorization server include:
 | `name` | String | The client name. | None |
 | `oauth2ClientId` | String | The unique identifier issued to a registered OAuth2 client by the authorization server. | None |
 | `oauth2ClientSecret` | String | Confidential shared secret between the client and the authorization server, used to verify the identity of the client. | None |
-| `redirectUri` | String | The URL on client's site where users will be sent after authorization. | None |
+| `oauth2ClientType` | String | The type of clients that use different OAuth2 grant type. GPII supports 4 type of clients: <br>`nativeApp`: native GPII apps that use OAuth2 resource owner password credentials grant.<br>`webApp`: web applications.<br>`thirdPartyApp`: third party applications that GPII users can define privacy settings for. <br>`clientCredentialsApp`: applications that use OAuth2 client credentials grant to add new preferences on behalf of users, such as First Discovery Tool.| Must be one of these values: "nativeApp", "webApp", "thirdPartyApp", "clientCredentialsApp".|

```

Comment:
Cindy and I chatted about `oauth2CientType` and also about the associated document types for capturing authorizations. Below are our notes.

In the current model, we have the following client types and associated authorization document types:

* nativeApp
    * OAuth2 Resource Owner Password Credentials grant type
    * For example: the GPII App
    * Authorization document type: resourceOwnerToken
* clientCredentialsApp
    * OAuth2 Client Credentials grant type
    * For example: First Discovery Tool
    * Authorization document type: clientCredentialsToken
* webApp
    * OAuth2 Authorization Code grant type
    * For example: no current use case (used in C4A to integrate web applications such as Easit4All)
    * Authorization document type: authDecision (shared with onboarded solutions below)
* thirdPartyApp
    * "Onboarded solutions", not using any OAuth2 grant type, integrated through the GPII Local Flow Manager
    * For example: Windows Magnifier
    * Authorization document type: authDecision (shared with web apps above)

We discussed renaming the existing values for `oauth2ClientType` and authorization document types; and splitting the `authDecision` authorization document type into 2. To have symmetry between the client types and authorization document types -- 1 to 1 mapping.

* Client types:
    * nativeApp -> oauth2ResourceOwnerPasswordCredentials
    * clientCredentialsApp -> oauth2ClientCredentials
    * webApp -> oauth2AuthCode
    * thirdPartyApp -> onboardedSolution
* Authorization document types:
    * resourceOwnerToken -> resourceOwnerPasswordCredentialsAuthorization
    * clientCredentialsToken -> clientCredentialsAuthorization
    * authDecision ->
        * authCodeAuthorization
        * onboardedSolutionAuthorization

("ResourceOwnerPasswordCredentials" is very long and it may be more reasonable to just use "ResourceOwner" if no ambiguity is introduced.)